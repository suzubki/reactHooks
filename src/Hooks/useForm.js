import { useState } from 'react'

// Recibe un objeto y retorna un nuevo objeto 'Values' que se genera al hacer un onChange
// a un input cualquiera. El reset() nos permite limpiar el campo

export const useForm = ( initialState = {} ) => {

    const [values, setValues] = useState(initialState)

    const reset = () => { 
        setValues(initialState)
     }

    const handleInputChange = ({ target }) => {

        setValues({
            ...values,
            [ target.name ]: target.value
        })
     }

  return [ values, handleInputChange, reset]
}
