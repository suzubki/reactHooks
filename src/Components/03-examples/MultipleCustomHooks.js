import React from 'react'
import { useFetch } from '../../Hooks/useFetch'
import { useCounter } from '../../Hooks/useCounter'

export const MultipleCustomHooks = () => {

  const { counter, increment } = useCounter(1)

  const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`)

  const { author, quote } = !!data && data[0]

  return (
    <div style={{ padding: "70px" }} >
      <h1>Breaking bad quotes</h1>
      <hr />


      { 
        loading ? 
          <div className='alert alert-info text-center' >
            Loading...
          </div>
        : 
          <figure className="text-end">
            <blockquote className="blockquote">
              <p> {quote} </p>
            </blockquote>
            <figcaption className="blockquote-footer">
              {author}
            </figcaption>
          </figure>
      }

        <button
          onClick={increment}
          className="btn btn-primary"
        >
          +1
        </button>


    </div>
  )
}
