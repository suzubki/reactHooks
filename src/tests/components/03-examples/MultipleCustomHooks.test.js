import { MultipleCustomHooks } from "../../../Components/03-examples/MultipleCustomHooks";
import { shallow } from "enzyme";
import { useFetch } from "../../../Hooks/useFetch";

jest.mock("../../../Hooks/useFetch");

describe("MultipleCustomHooks", () => {
    test("... snapshot", () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null,
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();
    });

    test("should return info", () => {
        useFetch.mockReturnValue({
            data: [
                {
                    author: "Darwin",
                    quote: "Hola darwin...",
                },
            ],
            loading: false,
            error: null,
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        console.log(wrapper.html());
    });
});
