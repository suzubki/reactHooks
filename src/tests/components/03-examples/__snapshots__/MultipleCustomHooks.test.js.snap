// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleCustomHooks ... snapshot 1`] = `
<div
  style={
    Object {
      "padding": "70px",
    }
  }
>
  <h1>
    Breaking bad quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    +1
  </button>
</div>
`;
